lib/checker_jobs/dsl.rb

```diff
@@ -40,10 +41,20 @@ def ensure_fewer(name, options = {}, &block)
   # Private API
   #
 
-  def notification_target
-    raise CheckerJobs::MissingNotificationTarget, self.class unless defined?(@notification_target)
+  def notifier
+    raise CheckerJobs::MissingNotifier, self.class unless defined?(@notifier)
+    @notifier
+  end
+
+  def notifier_options
+    raise CheckerJobs::MissingNotifier, self.class unless defined?(@notifier)
+
+    notifier_class = CheckerJobs.configuration.notifier_class(notifier)
+    unless notifier_class.valid_notify_options?(@notifier_options)
+      raise CheckerJobs::InvalidNotifierOptions
+    end
```

Comment:
Should we do that during the `#notify``call rather than each time we access the options?

Also I wonder about this not being the responsibility of the DSL. What if we simply move that validation to the `CheckerJobs::Notifiers::ActionMailer#initialize` method?